src/App.js

```diff
@@ -1,83 +1,129 @@
-};
+	color: 'blue'
+}
 
+// Show user info
 const UserInfo = props => {
-    const user = props.user;
-
-    return (
-        <div>
-            <img src={user.picture.large} />
-            <div>
-                {user.name.first + ' ' + user.name.last}
-            </div>
-        </div>
-    );
-};
+	const user = props.user;
+
+	return (
+		<div>
+			<img src={user.picture.large} />
+
+			<div>
+				{user.name.first + ' ' + user.name.last}
+			</div>
+		</div>
+	)
+}
+
+// List Dog and Cat people
+const UserList = props => {
+	const users = props.users;
+	const removeUser = props.handler;
+
+	return (
+		<div>
+			<h2>
+				{props.list} People
+			</h2>
+
+			<ul>
+				{
+					users.map(function(user, index) {
+						return (
+							<li>
+								<UserInfo user={user} />
+								<a onClick={() => removeUser(index, props.list)}>
+									X
+								</a>
+							</li>
+						)
+					})
+				}
+			</ul>
+		</div>
+	)
+}
 
 class App extends Component {
-    state = {
-        user: null,
-        dogPeople: [],
-        catPeople: []
-    };
-
-    getUserData = () => {
-        fetch('https://randomuser.me/api/')
-            .then(data => data.json())
-            .then(data => {
-                this.setState({
-                    user: data.results[0]
-                });
-            })
-            .catch(e => {
-                this.setState({error: e});
-            });
-    };
-
-    handlePetSelect = pet => {
-        /**
-         * Create two lists of dog people, cat people,
-         * show lists anywhere,
-         * reuse UserInfo component,
-         * allow deleting individual user from dogPeople/catPeople list
-         */
-        console.log(pet);
-
-        this.getUserData();
-    };
-
-    componentDidMount() {
-        this.getUserData();
-    }
-
-    render() {
-        const {user} = this.state;
-
-        if (!user) {
-            return <div>No User Found</div>;
-        }
-
-        return (
-            <div>
-                <h3 onClick={() => this.handlePetSelect('dog')} style={dogStyle}>
-                    Dog Person
-                </h3>
-                <h3 onClick={() => this.handlePetSelect('cat')} style={catStyle}>
-                    Cat Person
-                </h3>
-                <UserInfo user={user} />
-            </div>
-        );
-    }
+
+	state = {
+		number: 25,
+		dogPeople: [],
+		catPeople: [],
+	};
+
+	getUserData = () => {
+		fetch('https://randomuser.me/api/')
+			.then(data => data.json())
+			.then(data => {
+				this.setState({
+					user: data.results[0]
+				});
+			});
+	};
+
+	handlePetSelect = pet => {
+		// Log user into pet array
+		this.setState({
+			[pet + 'People']: this.state[pet + 'People'].concat({
+				name: this.state.user.name,
+				picture: this.state.user.picture,
+				pet: pet,
+			})
+		});
+
+		// Reset to new user
+		this.getUserData();
+	}
+
+	removeUser = (index, list) => {
+		let newData = this.state[list + 'People'].slice(); //copy array
```

Comment:
A simpler, more functional way of doing this might just be...
```js
const listKey = list + 'People';
this.setState({
    [listKey]: this.state[listKey].filter((value, i) => i !==  index)
})
```